tests/integration/rdb.tcl

```diff
@@ -198,3 +198,73 @@ test {client freed during loading} {
         exec kill [srv 0 pid]
     }
 }
+
+start_server {overrides {save ""}} {
+    test {Test child sending COW info} {
+        r config set rdb-key-save-delay 200
+
+        # populate the db
+        set rd [redis_deferring_client 0]
+        set size 4096
+        set cmd_count 10000
+        for {set k 0} {$k < $cmd_count} {incr k} {
+            $rd set key$k [string repeat A $size]
+        }
+
+        for {set k 0} {$k < $cmd_count} {incr k} {
+            catch { $rd read }
+        }
+
+        $rd close
+
+        # start background rdb save
+        r bgsave
+
+        # choose number of COW info reports:
+        # - accurate mode, 10 attempts should cover complete bgsave process, till the child will be exited.
+        # - no accutate mode, 2 attempts is enough to see and verify that COW info is reported
+        if {$::accurate} {
+            set attempts 10
+        } else {
+            set attempts 2
+        }
+
+        for {set j 0} {$j < $attempts} {incr j} {
+            # take a sample before writing new data to the server
+            set cow_info [s rdb_last_cow_size]
+            if {$::verbose} {
+                puts "COW info before copy-on-write: $cow_info"
+            }
+
+            # trigger copy-on-write
+            r setrange key$j 0 [string repeat B $size]
+
+            # wait to see that copy-on-write info updated
+            wait_for_condition 8 1000 {
+                [s rdb_last_cow_size] >= $cow_info + $size
+            } else {
+                # if the bgsave completed, maybe the last SETRANGE happened after the child already
+                # exited, which means no copy-on-right triggered and no COW info reported
+                if { [s rdb_bgsave_in_progress] == 0 } {
+                    if {$::verbose} {
+                        puts "bgsave completed"
+                    }
+                    return;
+                }
```

Comment:
i find this pattern a bit odd.
i'd prefer to move this to outside the loop.
i imagine the following changes:
- change the attempts loop to a while (not for)
- in `!::accurate` we'll break after 2 cycles. otherwise the loop runs forever.
- when the loop exists, if the fork child is no longer in progress (which will only be in the accurate mode), we check the final report